core/database/foxx/api/repo_router_new.js

```diff
@@ -0,0 +1,283 @@
+"use strict";
+
+const createRouter = require("@arangodb/foxx/router");
+const router = createRouter();
+const joi = require("joi");
+const { db: g_db } = require("@arangodb");
+const g_lib = require("./support");
+const { RepositoryType } = require("./repository/types");
+const { createRepositoryByType } = require("./repository/factory");
+const { RepositoryOps } = require("./repository/operations");
+
+module.exports = router;
+
+// Helper to clean repository data
+const cleanRepoData = (repo) => {
+    const { _id, _key, _rev, ...data } = repo.data;
+    return { ...data, id: _id };
+};
+
+// List repositories
+router
+    .get("/list", (req, res) => {
+        const client = req.queryParams.client
+            ? g_lib.getUserFromClientID(req.queryParams.client)
+            : null;
+
+        if (req.queryParams.all && client && !client.is_admin) {
+            throw g_lib.ERR_PERM_DENIED;
+        }
+
+        const filter = {
+            ...(req.queryParams.type && { type: req.queryParams.type }),
+            ...(!req.queryParams.all && client && { admin: client._id }),
+        };
+
+        const listResult = RepositoryOps.list(filter);
+        if (!listResult.ok) throw [listResult.error.code, listResult.error.message];
+
+        res.send(listResult.value.map(cleanRepoData));
+    })
+    .queryParam("client", joi.string().allow("").optional(), "Client ID")
+    .queryParam("details", joi.boolean().optional(), "Show additional record details")
+    .queryParam("all", joi.boolean().optional(), "List all repos (requires admin)")
+    .queryParam(
+        "type",
+        joi.string().valid("globus", "metadata_only").optional(),
+        "Filter by repository type",
+    )
+    .summary("List repositories")
+    .description("List repositories administered by the client.");
+
+// View repository
+router
+    .get("/view", (req, res) => {
+        try {
+            const findResult = RepositoryOps.find(req.queryParams.id);
+            if (!findResult.ok) throw [findResult.error.code, findResult.error.message];
+
+            const repo = cleanRepoData(findResult.value);
+            repo.admins = g_db.admin
+                .byExample({ _from: repo.id })
+                .toArray()
+                .map((admin) => admin._to);
+
+            res.send([repo]);
+        } catch (e) {
+            g_lib.handleException(e, res);
+        }
+    })
+    .queryParam("id", joi.string().required(), "Repository ID")
+    .summary("View repository")
+    .description("View repository details.");
+
+// Create repository
+router
+    .post("/create", (req, res) => {
```

Comment:
In the old method there were several calls to 

procInputParam 

for different fields, this validation piece seems to be missing, I don't see an equivalent in the code flows in this route or a call to that method.

That function seems to kind of do some custom schema validation and conversion checks.